File: frontend/src/services/history.js
import axios from 'axios';

const API_URL = 'http://localhost:8000';

export const HistoryService = {
    // Save prediction history
    async savePrediction(inputData, prediction, probabilities, userName) {
        try {
            const response = await axios.post(`${API_URL}/history/save`, {
                input_data: inputData,
                prediction: prediction,
                probabilities: probabilities,
                user_name: userName
            });
            return response.data;
        } catch (error) {
            console.error('Error saving prediction:', error);
            throw error;
        }
    },

    // Get prediction history
    async getHistory(limit = 10) {
        try {
            const response = await axios.get(`${API_URL}/history/list`, {
                params: { limit }
            });
            return response.data;
        } catch (error) {
            console.error('Error fetching history:', error);
            throw error;
        }
    }
};

export default HistoryService;

File: backend/app/history/history.py
from fastapi import APIRouter, HTTPException
from typing import Dict, List
from pydantic import BaseModel
from app.database.history_db import HistoryDB

class PredictionHistory(BaseModel):
    input_data: Dict
    prediction: int
    probabilities: List[float]
    user_name: str = None

router = APIRouter(
    prefix="/history",
    tags=["history"]
)

@router.post("/save")
async def save_prediction(history_data: PredictionHistory):
    try:
        await HistoryDB.save_prediction(
            history_data.input_data,
            history_data.prediction,
            history_data.probabilities,
            history_data.user_name
        )
        return {"message": "Prediction saved successfully"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/list")
async def get_history(limit: int = 10):
    try:
        history = await HistoryDB.get_history(limit)
        return history
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))